pkg/storegateway/batch_series.go

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package storegateway
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb/chunks"
+	"github.com/prometheus/prometheus/tsdb/hashcache"
+
+	"github.com/grafana/mimir/pkg/storage/sharding"
+	"github.com/grafana/mimir/pkg/storegateway/storepb"
+)
+
+type batchedSeriesSet struct {
+	ctx      context.Context
+	postings []storage.SeriesRef
+
+	batchSize int
+	i         int // where we're at in the labels and chunks
+	iOffset   int // if i==0, which index in postings does this correspond to
```

Comment:
[nit] I found a bit difficult to understand what these are from naming. I have a couple of suggestions. A bit of verbosity may help.

```suggestion
	currBatchIdx         int // where we're at in the labels and chunks
	nextBatchPostingsOffset   int // if i==0, which index in postings does this correspond to
```